src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxNode.cs

```diff
@@ -184,43 +146,49 @@ public override GreenNode GetTrailingTriviaCore()
             return this.GetTrailingTrivia();
         }
 
-        public override string ToString()
-        {
-            var sb = PooledStringBuilder.GetInstance();
-            var writer = new System.IO.StringWriter(sb.Builder, System.Globalization.CultureInfo.InvariantCulture);
-            this.WriteTo(writer, leading: false, trailing: false);
-            return sb.ToStringAndFree();
-        }
-
-        public override string ToFullString()
-        {
-            var sb = PooledStringBuilder.GetInstance();
-            var writer = new System.IO.StringWriter(sb.Builder, System.Globalization.CultureInfo.InvariantCulture);
-            this.WriteTo(writer, leading: true, trailing: true);
-            return sb.ToStringAndFree();
-        }
-
         public abstract TResult Accept<TResult>(CSharpSyntaxVisitor<TResult> visitor);
 
         public abstract void Accept(CSharpSyntaxVisitor visitor);
 
         internal virtual DirectiveStack ApplyDirectives(DirectiveStack stack)
         {
-            if (this.ContainsDirectives)
+            return ApplyDirectives(this, stack);
+        }
+
+        internal static DirectiveStack ApplyDirectives(GreenNode node, DirectiveStack stack)
+        {
+            if (node.ContainsDirectives)
             {
-                for (int i = 0, n = this.SlotCount; i < n; i++)
+                for (int i = 0, n = node.SlotCount; i < n; i++)
                 {
-                    var child = this.GetSlot(i);
+                    var child = node.GetSlot(i);
                     if (child != null)
                     {
-                        stack = ((CSharpSyntaxNode)child).ApplyDirectives(stack);
+                        stack = ApplyDirectivesToListOrNode(child, stack);
                     }
                 }
             }
 
             return stack;
         }
 
+        internal static DirectiveStack ApplyDirectivesToListOrNode(GreenNode listOrNode, DirectiveStack stack)
+        {
+            // If we have a list of trivia, then that node is not actually a CSharpSyntaxNode.
```

Comment:
Some internal nodes are no longer `CSharpSyntaxNode`s? Are all callers expecting that?